### DISCUSSION TITLE
Free Memory changes a lot

### DISCUSSION THREAD

**Discussion by kglasnapp** [wpilibsuite/2024Beta#45] [Q&A]
I plotted free memory on the Shuffle Board and it changes a lot even when the robot is disabled. See attached plot.
![Screenshot 2023-12-15 033439](https://github.com/wpilibsuite/2024Beta/assets/4080114/e8bf7ebf-0938-4348-b68e-e7941d791ef9)

**Comment by ThadHouse** [comment]
Thats how Java works. It will allocate up to a point, and then run a garbage collection. It will keep doing this forever, because objects will be allocated.

**Comment by sciencewhiz** [comment]
If I'm reading that right, it's allocating 125 megabytes in 5 seconds. That certainly seems excessive. Does this occur on an empty project, or only with your full code? There are some tips for reducing memory usage here: https://docs.wpilib.org/en/latest/docs/software/basic-programming/java-gc.html

**Comment by PeterJohnson** [comment]
Also, free memory is going to fluctuate due to OS caching (free memory is wasted memory).  Available memory is a better metric.

**Reply by kglasnapp** [reply]
This occurs on a very simple robot with only drive motors. It occurs in Disabled and Enable Modes.  125 megs seems very excessive to me.

How do you access available memory?

**Reply by sciencewhiz** [reply]
Both free and available are in top and /proc/meminfo

How did you access free memory?
